spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/types/TypeConverter.scala

```diff
@@ -409,6 +420,35 @@ object TypeConverter {
     }
   }
 
+  private val LocalDateTypeTag = TypeTag.synchronized {
+    implicitly[TypeTag[LocalDate]]
+  }
+
+  implicit object LocalDateConverter extends NullableTypeConverter[LocalDate] {
+    def targetTypeTag = LocalDateTypeTag
+    val dateRegx = """(\d\d\d\d)-(\d\d)-(\d\d)""".r
+
+    def addTimeZoneOffset(millis: Long) =  millis + defaultTimezone.getOffset(millis)
+
+    def convertPF = {
+      case x: LocalDate => x
+      case x: String => x match { case dateRegx(year,month,day) =>
```

Comment:
Not sure what you mean here? ( i understand the formatting mistake but not the regex immediately match?, Do you mean case dateRegex instead of case String?